### PR TITLE
Step 1:Initial setup

### PR DESCRIPTION
**Pull request by vhoreho**

### CODE REVIEW

File: components/ActiveLink.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
+// @ts-ignore
```
Comment (Ilya-Valasiuk) [inline components/ActiveLink.tsx:1]: why we need ts-ignore?

File: components/ActiveLink.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import { useRouter } from 'next/router'
+// @ts-ignore
+import Link, { LinkProps } from 'next/link'
+import React, { useState, useEffect, ReactElement, Children } from 'react'
```
Comment (Ilya-Valasiuk) [inline components/ActiveLink.tsx:5]: please import react first in order

File: components/ActiveLink.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+type ActiveLinkProps = LinkProps & {
+    children: ReactElement
+    activeClassName: string
+}
+
+const ActiveLink = ({
```
Comment (Ilya-Valasiuk) [inline components/ActiveLink.tsx:12]:
```
const ActiveLink: FunctionalComponent<Props> = ({ activeClassName, ...props }) => {
```

File: components/ActiveLink.tsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    return (
+        <Link {...props}>
+            {React.cloneElement(child, {
+                className: className || null,
+            })}
+        </Link>
+    )
+}
+
+export default ActiveLink
```
Comment (Ilya-Valasiuk) [inline components/ActiveLink.tsx:65]: no default exports please - https://docs.google.com/document/d/1ez__g_pUi-T5aRnsBxumdu-stTuk6xZ6PtZroAd5BeY/edit#heading=h.e56r2bqdnrmg

File: components/Header/Burger.tsx [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import styles from '../../styles/components/header/Burger.module.scss'
+
+export const Burger = ({open, setOpen}) => {
+    return (
+        <button className={`${styles.button} ${open ? 'active' : ''}`} onClick={() => setOpen(!open)}>
```
Comment (Ilya-Valasiuk) [inline components/Header/Burger.tsx:5]: use [classnames](https://www.npmjs.com/package/classnames) lib

File: components/Header/Menu.tsx [hunk 6]
```diff
@@ -0,0 +1,28 @@
+import styles from '../../styles/components/header/Menu.module.scss';
+// @ts-ignore
```
Comment (Ilya-Valasiuk) [inline components/Header/Menu.tsx:2]: ?

File: components/Header/Menu.tsx [hunk 7]
```diff
@@ -0,0 +1,28 @@
+import styles from '../../styles/components/header/Menu.module.scss';
```
Comment (Ilya-Valasiuk) [inline components/Header/Menu.tsx:1]: https://docs.google.com/document/d/1b1Bt3sM4cKvmYgmNka41vWqeHzUZ4no2Jnb5tjZxd10/edit#

File: components/layout.tsx [hunk 8]
```diff
@@ -0,0 +1,19 @@
+import {ReactNode} from "react";
+// @ts-ignore
+import Header from "./Header/header.tsx";
+// @ts-ignore
+import Footer from "./footer.tsx";
+
+import styles from '../styles/common.module.scss'
+
+export default function Layout({children}: {children:ReactNode}) {
```
Comment (Ilya-Valasiuk) [inline components/layout.tsx:9]: do we need layout here or we can use just `_app.tsx`?

File: package.json [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+  "version": "0.1.0",
+  "private": true,
+  "scripts": {
+    "dev": "next dev",
+    "build": "next build",
+    "start": "next start",
+    "lint": "next lint"
+  },
+  "dependencies": {
+    "@types/uuid": "^8.3.4",
```
Comment (Ilya-Valasiuk) [inline package.json:12]: why types in dependencies?

File: pages/api/hello.js [hunk 10]
```diff
@@ -0,0 +1,5 @@
+// Next.js API route support: https://nextjs.org/docs/api-routes/introduction
+
+export default function handler(req, res) {
```
Comment (Ilya-Valasiuk) [inline pages/api/hello.js:3]: do we need this?

File: tsconfig.json [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+    "strict": false,
+    "forceConsistentCasingInFileNames": true,
+    "noEmit": true,
+    "incremental": true,
+    "esModuleInterop": true,
+    "module": "esnext",
+    "resolveJsonModule": true,
+    "isolatedModules": true,
+    "jsx": "preserve"
+  },
```
Comment (Ilya-Valasiuk) [inline tsconfig.json:20]: please use absolute imports and place all code to `src` folder - https://nextjs.org/docs/advanced-features/module-path-aliases

**Review by Ilya-Valasiuk — CHANGES_REQUESTED** [review]

File: components/ActiveLink.tsx (hunk 1 above)
Comment (vhoreho) [reply components/ActiveLink.tsx:1]: WebStorm linter swears

File: components/ActiveLink.tsx (hunk 2 above)
Comment (vhoreho) [reply components/ActiveLink.tsx:5]: Done

File: components/ActiveLink.tsx (hunk 3 above)
Comment (vhoreho) [reply components/ActiveLink.tsx:12]: Done

File: pages/api/hello.js (hunk 10 above)
Comment (vhoreho) [reply pages/api/hello.js:3]: Done

File: tsconfig.json (hunk 11 above)
Comment (vhoreho) [reply tsconfig.json:20]: done

File: package.json (hunk 9 above)
Comment (vhoreho) [reply package.json:12]: done

File: components/Header/Menu.tsx (hunk 7 above)
Comment (vhoreho) [reply components/Header/Menu.tsx:1]: done

File: components/Header/Menu.tsx (hunk 6 above)
Comment (vhoreho) [reply components/Header/Menu.tsx:2]: Deleted

File: components/Header/Burger.tsx (hunk 5 above)
Comment (vhoreho) [reply components/Header/Burger.tsx:5]: Done

File: components/ActiveLink.tsx (hunk 4 above)
Comment (vhoreho) [reply components/ActiveLink.tsx:65]: done

File: src/components/layout/Layout.tsx [hunk 12]
```diff
@@ -0,0 +1,15 @@
+import {ReactNode} from "react";
+import {Header} from "./header/Header";
+import {Footer} from "./footer/Footer";
+
+export function Layout({children}: {children:ReactNode}) {
```
Comment (Ilya-Valasiuk) [inline src/components/layout/Layout.tsx:5]: not needed

File: src/utils/fetcher.ts [hunk 13]
```diff
@@ -0,0 +1,5 @@
+export const fetcher = async (category) => {
+    const response = await fetch(`https://api.nytimes.com/svc/topstories/v2/${category}.json?api-key=ZGnYeMXFe2qj5uVKnmdIF36750qfoiG0`);
```
Comment (Ilya-Valasiuk) [inline src/utils/fetcher.ts:2]: api token should be in env

Comment (Ilya-Valasiuk) [inline src/utils/fetcher.ts:2]: api path should be in constants/api.ts file

File: src/utils/fetcher.ts [hunk 14]
```diff
@@ -0,0 +1,5 @@
...
+    const data = await response.json();
```
Comment (Ilya-Valasiuk) [inline src/utils/fetcher.ts:3]: try/catch

File: src/components/layout/header/burger/Burger.module.scss [hunk 15]
```diff
@@ -0,0 +1,38 @@
+@import '../../../../styles/mixins';
```
Comment (Ilya-Valasiuk) [inline src/components/layout/header/burger/Burger.module.scss:1]: @import 'styles/mixins';

File: src/components/layout/header/Header.module.scss [hunk 16]
```diff
@@ -0,0 +1,85 @@
+@import "../../../styles/variables";
```
Comment (Ilya-Valasiuk) [inline src/components/layout/header/Header.module.scss:1]: file name `styles.module.scss`

File: src/components/layout/header/Header.module.scss [hunk 17]
```diff
@@ -0,0 +1,85 @@
...
+@import "../../../styles/mixins";
```
Comment (Ilya-Valasiuk) [inline src/components/layout/header/Header.module.scss:2]: @import "styles/mixins";

File: src/components/layout/header/Header.module.scss [hunk 18]
```diff
@@ -0,0 +1,85 @@
...
+
+.header {
+  width: 100%;
+  gap: 10px;
+  background: #FBFBFB;
```
Comment (Ilya-Valasiuk) [inline src/components/layout/header/Header.module.scss:7]: all colors, all shadows should be in `styles/variables.scss`

File: src/components/ActiveLink.tsx [hunk 19]
```diff
@@ -0,0 +1,61 @@
+import React, {useState, useEffect, ReactElement, Children, FunctionComponent} from 'react'
```
Comment (Ilya-Valasiuk) [inline src/components/ActiveLink.tsx:1]: `components/common/active-link/ActiveLink.tsx`;

File: src/components/common/footer/styles.module.scss [hunk 20]
```diff
@@ -0,0 +1,70 @@
+@import "styles/mixins";
+
+.footer {
+  color: #7A7A7A;
```
Comment (Ilya-Valasiuk) [inline src/components/common/footer/styles.module.scss:4]: colors and shadows should be from `styles/variables`

File: src/components/common/footer/styles.module.scss [hunk 21]
```diff
@@ -0,0 +1,70 @@
...
+  font-size: 14px;
+}
+
+.container {
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  flex-wrap: wrap;
+  border-top: 1px solid #D1D1D1;
```
Comment (Ilya-Valasiuk) [inline src/components/common/footer/styles.module.scss:13]: the same here

File: src/components/common/footer/styles.module.scss [hunk 22]
```diff
@@ -0,0 +1,70 @@
...
+.nav .link:not(:last-of-type) {
+  margin-right: 32px;
+  @include for-tablet-landscape-up {
+    margin-right: 0;
+    margin-bottom: 10px;
+  }
+}
+
+.link:hover {
+  color: #161616;
```
Comment (Ilya-Valasiuk) [inline src/components/common/footer/styles.module.scss:44]: the same

File: src/components/common/footer/styles.module.scss [hunk 23]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+.footer__socials {
+  list-style-type: none;
+  display: flex;
+  padding: 0;
+  margin: 0;
+  @include for-tablet-landscape-up {
+    margin: 1.5rem 0;
```
Comment (Ilya-Valasiuk) [inline src/components/common/footer/styles.module.scss:53]: is there reason why we mix rem with px ?

File: src/components/common/footer/styles.module.scss [hunk 24]
```diff
@@ -0,0 +1,70 @@
...
+  }
+}
+
+.socials {
+  &__item:not(:last-of-type) {
+    margin-right: 13px;
+
+  }
+  &_fill {
+    fill: #7A7A7A;
```
Comment (Ilya-Valasiuk) [inline src/components/common/footer/styles.module.scss:63]: the same

File: src/components/common/footer/styles.module.scss [hunk 25]
```diff
@@ -0,0 +1,70 @@
...
+  }
+  &_fill:hover {
+    fill: #161616;
```
Comment (Ilya-Valasiuk) [inline src/components/common/footer/styles.module.scss:66]: the same

File: src/components/common/header/Header.tsx [hunk 26]
```diff
@@ -0,0 +1,47 @@
+import {useState} from "react";
+import Link from "next/link";
+import Image from "next/image";
+
+import {ActiveLink} from "../active-link/ActiveLink";
```
Comment (Ilya-Valasiuk) [inline src/components/common/header/Header.tsx:5]: * formatting
* absolute path

```
import { ActiveLink } from "components/common/active-link/ActiveLink";
```

File: src/components/common/header/Header.tsx [hunk 27]
```diff
@@ -0,0 +1,47 @@
...
+import {Burger} from "./burger/Burger";
+import {Menu} from "./menu/Menu";
+
+import styles from './styles.module.scss';
+
+import logo from '../../../../public/logo.svg';
```
Comment (Ilya-Valasiuk) [inline src/components/common/header/Header.tsx:11]: import order, absolute import ...

File: src/components/common/header/burger/Burger.tsx [hunk 28]
```diff
@@ -0,0 +1,17 @@
+import classNames from "classnames";
```
Comment (Ilya-Valasiuk) [inline src/components/common/header/burger/Burger.tsx:1]: this should be in `common/header/components/burger/...`

File: src/components/common/header/burger/styles.module.scss [hunk 29]
```diff
@@ -0,0 +1,38 @@
...
+  }
+
+  &:focus {
+    outline: none;
+  }
+
+  div {
+    width: 2rem;
+    height: 0.25rem;
+    background: #161616;
```
Comment (Ilya-Valasiuk) [inline src/components/common/header/burger/styles.module.scss:26]: variable

File: src/components/common/header/burger/styles.module.scss [hunk 30]
```diff
@@ -0,0 +1,38 @@
...
+  @include for-tablet-landscape-up {
+    display: flex;
+  }
+
+  &:focus {
+    outline: none;
+  }
+
+  div {
+    width: 2rem;
```
Comment (Ilya-Valasiuk) [inline src/components/common/header/burger/styles.module.scss:24]: rem with px, please use 1 approach

File: src/components/common/header/burger/styles.module.scss [hunk 31]
```diff
@@ -0,0 +1,38 @@
...
+    border-radius: 10px;
+    transition: all 0.3s linear;
+    position: relative;
+    transform-origin: 1px;
+  }
+}
+
+.button.active {
+  div {
+    background: #ffffff;
```
Comment (Ilya-Valasiuk) [inline src/components/common/header/burger/styles.module.scss:36]: variable

File: src/components/common/header/Header.tsx [hunk 32]
```diff
@@ -0,0 +1,47 @@
...
+            <Link href='/' >
+                <a className={styles.logo}>
+                    <Image
+                        alt='Best News'
+                        src={logo}
+                    />
+                </a>
+            </Link>
+            <nav className={styles.nav}>
+                <ActiveLink href="/" activeClassName={styles.active}>
```
Comment (Ilya-Valasiuk) [inline src/components/common/header/Header.tsx:27]: Why we need `ActiveLink` at all?

Why we can't define what is active here and do something like this `<Link className={isActive ? "styles.active" : ""}`

File: src/components/common/header/burger/Burger.tsx [hunk 33]
```diff
@@ -0,0 +1,17 @@
...
+
+import styles from './styles.module.scss'
+
+export const Burger = ({open, setOpen}) => {
+    const btnClass = classNames(styles.button, {
```
Comment (Ilya-Valasiuk) [inline src/components/common/header/burger/Burger.tsx:6]: do we need this variable at all? 

```
<button className={classNames(styles.button, { 'active': isOpen })} onClick={() => setOpen(!open)}>
```

File: src/components/common/header/styles.module.scss [hunk 34]
```diff
@@ -0,0 +1,85 @@
+@import "styles/variables";
+@import "styles/mixins";
+
+.header {
+  width: 100%;
+  gap: 10px;
+  background: #FBFBFB;
```
Comment (Ilya-Valasiuk) [inline src/components/common/header/styles.module.scss:7]: variable

File: src/components/common/header/styles.module.scss [hunk 35]
```diff
@@ -0,0 +1,85 @@
...
+  box-shadow: 0 2px 10px rgba(22, 22, 22, 0.1);
```
Comment (Ilya-Valasiuk) [inline src/components/common/header/styles.module.scss:8]: variable

File: src/constants/api.ts [hunk 36]
```diff
@@ -0,0 +1 @@
+export const API_URL:string = 'https://api.nytimes.com/svc/topstories/v2';
```
Comment (Ilya-Valasiuk) [inline src/constants/api.ts:1]: you do not need to cast variable to string as it already string.

+ it should come from `process.env.BASE_PATH` variable.

**Review by Ilya-Valasiuk — CHANGES_REQUESTED** [review]

**Review by Ilya-Valasiuk — APPROVED** [review]

**Pull request merged**